packages/dart_firebase_admin/lib/src/utils/jwt.dart

```diff
@@ -96,9 +99,51 @@ class PublicKeySignatureVerifier implements SignatureVerifier {
   final KeyFetcher keyFetcher;
 
   @override
-  Future<bool> verify(String token) {
-    throw UnimplementedError();
-    // verifyJwtSignature(token);
+  Future<void> verify(String token) async {
+    try {
+      final jwt = JWT.decode(token);
+      final kid = jwt.header?['kid'] as String?;
+
+      if (kid == null) {
+        throw JwtError(
+          JwtErrorCode.noKidInHeader,
+          noKidInHeaderErrorMessage,
+        );
+      }
+
+      final publicKeys = await keyFetcher.fetchPublicKeys();
+      final publicKey = publicKeys[kid];
+
+      if (publicKey == null) {
+        throw JwtError(
+          JwtErrorCode.noMatchingKid,
+          moMatchingKidErrorMessage,
+        );
+      }
+      JWT.verify(
+        token,
+        RSAPublicKey.cert(publicKey),
+      );
+    } on JWTExpiredException {
+      throw JwtError(
+        JwtErrorCode.tokenExpired,
+        'The provided token has expired. Get a fresh token from your client app and try again.',
+      );
+    } on JWTException catch (e) {
+      if (e.message.startsWith(jwtCallbackErrorPrefix)) {
```

Comment:
We shouldn't depend on the english message.

Reply:
And then, how about simplifying this error handling for now?

Just like:

```dart
...

} on JWTExpiredException {
      throw JwtError(
        JwtErrorCode.tokenExpired,
        'The provided token has expired. Get a fresh token from your client app and try again.',
      );
    } on JWTException catch (e) {
    // Just throw JwtErrorCode.unknown for now.
    throw JwtError(JwtErrorCode.unknown, e.message);
}

...


enum JwtErrorCode {
  invalidArgument('invalid-argument'),
   ...

   // add this enum field
  unknown('unknown');

  ...
}


```

Reply:
Sure. I shouldn't have done that indeed.

Reply:
@tkfinitefield 

Hello! 
I would appreciate it if you could take a moment to check these comments and share your thoughts.